src/devices/Bmxx80/Bmx280Base.cs

```diff
@@ -0,0 +1,291 @@
+            }
+
+            // Read pressure data.
+            byte msb = Read8BitsFromRegister((byte)Bmx280Register.PRESSUREDATA_MSB);
+            byte lsb = Read8BitsFromRegister((byte)Bmx280Register.PRESSUREDATA_LSB);
+            byte xlsb = Read8BitsFromRegister((byte)Bmx280Register.PRESSUREDATA_XLSB); // bits 7:4
+
+            //Combine the values into a 32-bit integer.
+            int t = (msb << 12) + (lsb << 4) + (xlsb >> 4);
+
+            //Convert the raw value to the pressure in Pa.
+            long pres = CompensatePressure(t);
+
+            //Return the temperature as a float value.
+            return (double)pres / 256;
+        }
+
+        /// <summary>
+        /// Calculates the altitude in meters from the specified sea-level pressure(in hPa).
+        /// </summary>
+        /// <param name="seaLevelPressure">Sea-level pressure in hPa.</param>
+        /// <returns>Height in meters from sea-level.</returns>
+        public async Task<double> ReadAltitudeAsync(double seaLevelPressure)
+        {
+            // Read the pressure first.
+            double pressure = await ReadPressureAsync();
+
+            // Convert the pressure to Hectopascals (hPa).
+            pressure /= 100;
+
+            // Calculate and return the altitude using the international barometric formula.
+            return 44330.0 * (1.0 - Math.Pow((pressure / seaLevelPressure), 0.1903));
+        }
+
+        /// <summary>
+        /// Get the current status of the device.
+        /// </summary>
+        /// <returns>The <see cref="DeviceStatus"/>.</returns>
+        public DeviceStatus ReadStatus()
+        {
+            var status = Read8BitsFromRegister((byte)Bmx280Register.STATUS);
+
+            // Bit 3.
+            var measuring = ((status >> 3) & 1) == 1;
+
+            // Bit 0.
+            var imageUpdating = (status & 1) == 1;
+
+            return new DeviceStatus
+            {
+                ImageUpdating = imageUpdating,
+                Measuring = measuring
+            };
+        }
+
+        /// <summary>
+        /// Sets the IIR filter mode.
+        /// </summary>
+        /// <param name="filteringMode">The <see cref="FilteringMode"/> to set.</param>
+        public void SetFilterMode(FilteringMode filteringMode)
+        {
+            byte current = Read8BitsFromRegister((byte)Bmx280Register.CONFIG);
+            current = (byte)((current & 0b_1110_0011) | (byte)filteringMode << 2);
+            _i2cDevice.Write(new[] { (byte)Bmx280Register.CONFIG, current });
+        }
+
+        /// <summary>
+        /// Sets the power mode to the given mode
+        /// </summary>
+        /// <param name="powerMode">The <see cref="Bmx280PowerMode"/> to set.</param>
+        public void SetPowerMode(Bmx280PowerMode powerMode)
+        {
+            byte status = Read8BitsFromRegister(_controlRegister);
+
+            // Clear last two bits.
+            status = (byte)(status & 0b_1111_1100);
```

Comment:
NIT: This could technically also be converted into a one-line as the other sets where you combine both bitwise operations.